source/byte_buf.c

```diff
@@ -405,28 +413,39 @@ bool aws_byte_cursor_eq_c_str_ignore_case(const struct aws_byte_cursor *cursor,
 }
 
 int aws_byte_buf_append(struct aws_byte_buf *to, const struct aws_byte_cursor *from) {
-    assert(from->ptr);
-    assert(to->buffer);
+    AWS_PRECONDITION(aws_byte_buf_is_valid(to));
+    AWS_PRECONDITION(aws_byte_cursor_is_valid(from));
+    if (!from->ptr || !to->buffer) {
+        return aws_raise_error(AWS_ERROR_INVALID_ARGUMENT);
+    }
 
     if (to->capacity - to->len < from->len) {
+        AWS_POSTCONDITION(aws_byte_buf_is_valid(to));
+        AWS_POSTCONDITION(aws_byte_cursor_is_valid(from));
         return aws_raise_error(AWS_ERROR_DEST_COPY_TOO_SMALL);
     }
 
     memcpy(to->buffer + to->len, from->ptr, from->len);
     to->len += from->len;
+    AWS_POSTCONDITION(aws_byte_buf_is_valid(to));
+    AWS_POSTCONDITION(aws_byte_cursor_is_valid(from));
     return AWS_OP_SUCCESS;
 }
 
 int aws_byte_buf_append_with_lookup(
     struct aws_byte_buf *AWS_RESTRICT to,
     const struct aws_byte_cursor *AWS_RESTRICT from,
     const uint8_t *lookup_table) {
-    AWS_PRECONDITION(from->ptr);
     AWS_PRECONDITION(aws_byte_buf_is_valid(to));
     AWS_PRECONDITION(aws_byte_cursor_is_valid(from));
     AWS_PRECONDITION(AWS_MEM_IS_READABLE(lookup_table, 256));
+    if (!from->ptr || !to->buffer) {
```

Comment:
Same here: is this additional to the other preconditions.

Reply:
Same answer too.